File: c++/Quicksort.cpp
// Quick Sort

#include<bits/stdc++.h>
#include<stdio.h>

using namespace std;

 void printArray(int arr[],int size)
 {
    for(int i=0;i<size;i++)
        cout<<arr[i]<<" ";
    cout<<endl;
 }

 int partition(int arr[],int low,int high)
 {
    int pivot = arr[low];
    int start= low+1;
    int end = high;
    int temp;
    do{
        while(arr[start] <= pivot)
        {
            start++;
        }
        while(arr[end]> pivot)
        {
            end--;
        } 
        if(start<end)
        {
            temp= arr[start];
            arr[start] = arr[end];
            arr[end]= temp;
        }
    }while(start<end);

    // Swap arr[low] and arr[end]
    temp = arr[low];
    arr[low]= arr[end];
    arr[end]=temp;
    return end;

 }

void quickSort(int arr[],int low,int high)
{
    int partiitionindex;
    if(low<high)
    {
        partiitionindex = partition(arr,low,high);
        quickSort(arr,low,partiitionindex-1);
        quickSort(arr,partiitionindex+1,high);
    }
}


int main()
{
    int n;
    cout<<"Enter the size of the array:";
    cin>>n;
    int A[n];
    cout<<"Enter "<<n<<" elements:";
    for(int i=0;i<n;i++)
    {
        cin>>A[i];
    }
    printArray(A,n);
    quickSort(A,0,n-1);
    printArray(A,n);
    return 0;
}

File: Python/quick_sort.py
def partition(array, low, high):
    pivot = array[high]
    i = low - 1
 
    for j in range(low, high):
        if array[j] <= pivot:
            i = i + 1
            (array[i], array[j]) = (array[j], array[i])
 
    (array[i + 1], array[high]) = (array[high], array[i + 1])
 
    return i + 1
 
def quickSort(array, low, high):
    if low < high:
        pi = partition(array, low, high)
 
        quickSort(array, low, pi - 1)
 
        quickSort(array, pi + 1, high)
 

n = int(input("Number of elements - "))
l = []

for i in range(n):
    a = int(input("Enter the element - "))
    l.append(a)
    
print("Unsorted Array")
print(l)
 
size = len(l)
 
quickSort(l, 0, size - 1)
 
print('Sorted Array in Ascending Order:')
print(l)
